### ISSUE TITLE
General code gardening

### ISSUE THREAD

**Issue by P3trur0** [freedomotic/freedomotic#270]
Hi there,

navigating in the source code I see a lot of commented lines. Is it worth to clean it a little by removing unnecessary commented lines?

The main advantage of this cleaning is to make the code easier to read, then more elegant and appealing for further contributors.

Opinions anyone?

**Comment by mcicolella** [comment]
@P3trur0 I agree. We should review the code, remove unnecessary comments and create new issues from "TODO" ones. Also please consider  https://sonarcloud.io/dashboard?id=com.freedomotic%3Afreedomotic

**Comment by P3trur0** [comment]
@mcicolella cool, I think that's important to have a cleaner source code.
May I take a look at this too?

**Comment by mcicolella** [comment]
@P3trur0 of course! Your support is very precious!

**Comment by mcicolella** [comment]
@P3trur0 For "Define and throw a dedicated exception instead of using a generic one" (RuntimeExceptions) issues reported by Sonarcloud can we use [FreedomoticRuntimeException](https://github.com/freedomotic/freedomotic/blob/master/framework/freedomotic-core/src/main/java/com/freedomotic/exceptions/FreedomoticRuntimeException.java) in any case for now?

**Comment by P3trur0** [comment]
@mcicolella yes indeed. You can consider `FreedomoticRuntimeException` as the root of the unchecked exceptions of Freedomotic.
